### PR TITLE
Generate props for typescript components

### PR DESCRIPTION
**Pull request by jschuler**
Using typedoc to parse ts files which spits out a json file. Read the json file in gatsby and use graphql to find the component. This will likely need to be adjusted when more TS components are added but should give us a good base to build upon.
Closes: #1033

### CODE REVIEW

File: packages/patternfly-4/react-core/src/tsconfig.dts.json [hunk 1]
```diff
@@ -8,16 +8,28 @@
     "declaration": true,
     "emitDeclarationOnly": true,
     "skipLibCheck": true,
-    "sourceMap": false, // allow sourcemap support
-    "lib": ["es2017", "dom"],
+    "sourceMap": false,
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/tsconfig.dts.json]: Comments are not allowed in json files

File: packages/patternfly-4/react-docs/src/components/componentDocs/componentDocs.js [hunk 2]
```diff
@@ -57,6 +58,7 @@ class ComponentDocs extends React.PureComponent {
     } = this.props;
     const makeDescription = html => ({ __html: html });
     const getDocGenInfo = name => data.allComponentMetadata.edges.find(edge => edge.node.displayName === name);
+    const getDocGenInfoTs = name => data.allTsDocsJson.edges[0].node.data.find(edge => edge.name === `${name}Props`);
```
Comment (jschuler) [inline packages/patternfly-4/react-docs/src/components/componentDocs/componentDocs.js]: Right now making the assumption to find the corresponding prop for a component by appending `Props` to the component name, e.g. Avatar => AvatarProps

File: packages/patternfly-4/react-integration/demo-app-ts/src/App.tsx [hunk 3]
```diff
@@ -77,7 +74,7 @@ class App extends Component {
           action={<AlertActionLink>Action Button</AlertActionLink>}
         />
         <Alert variant="success" title="Success notification title" />
-        <Select
+        {/* <Select
```
Comment (dgutride) [inline packages/patternfly-4/react-integration/demo-app-ts/src/App.tsx]: is select broken right now?

Comment (jschuler) [reply packages/patternfly-4/react-integration/demo-app-ts/src/App.tsx]: I must have had outdated react-core in my node_modules folder so it couldn't find those exports. I cleared and reinstalled and now is working. I reverted the change to this file

File: packages/patternfly-4/react-core/src/tsconfig.dts.json (hunk 1 above)
Comment (dlabaj) [reply packages/patternfly-4/react-core/src/tsconfig.dts.json]: How did this work before?

Comment (jschuler) [reply packages/patternfly-4/react-core/src/tsconfig.dts.json]: Gatsby is now reading json files in and parsing them so the newly added plugin doesn't like comments in json

Comment (jschuler) [reply packages/patternfly-4/react-core/src/tsconfig.dts.json]: https://stackoverflow.com/questions/244777/can-comments-be-used-in-json

**Pull request merged**